suse_migration_services/prechecks/kernels.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2022
+#
+# SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+"""Pre-Checks multiversion kernels
+
+Check to see if multiversion.kernels are enabled in /etc/zypp/zypp.conf and
+if there is more than one kernel currently installed. If detected a warning is reported
+as multiversion.kernels needs to be disabled and old kernels cleaned up before
+migrationg."""
+
+# import os
+import configparser
+import logging
+
+from suse_migration_services.defaults import Defaults
+from suse_migration_services.command import Command
+
+
+def multiversion_kernels():
+    """Check for the number of installed kernels"""
+    log = logging.getLogger(Defaults.get_migration_log_name())
+
+    config = configparser.ConfigParser()
+    config.read(Defaults.get_zypp_config_path())
+
+    current_settings = config.get('main', 'multiversion.kernels', fallback=None)
+
+    if not current_settings:
+        log.warning('Missing multiversion.kernels setting in zypp.conf. '
```

Comment:
what's the default behavior if the option is missing?

Reply:
The default in the cloud images is 

`multiversion.kernels = latest,latest-1,running`

the images are built this way so it would only be missing (removed or commented out)  if a customer did this change after the image was booted.

Reply:
got it. Thanks.

Reply:
Which means by default we ship a configuration that enables multi kernels, hmm. This also means we are giving users a configuration that is explicitly not supported for migration.

Last but not lease the images we deliver are not ready for DMS. We should change our images to address this issue. Please create a bug for images starting with SLES 12 SP4.